### PR TITLE
observe value spec on init query builer

### PR DESCRIPTION
**Pull request by MauricioUyaguari**
## Summary

Add observable to query lambda in query builder. 
Partially Addresses https://github.com/finos/legend-studio/issues/1037

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

File: .changeset/cold-doors-hope.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@finos/legend-query': patch
+---
+
+Observe ValueSpecification when initializing query builder
```
Comment (akphi) [inline .changeset/cold-doors-hope.md:5]:
```suggestion
Observe `ValueSpecification` when initializing query builder.
```

File: packages/legend-query/src/stores/QueryBuilderState.ts [hunk 2]
```diff
@@ -260,6 +263,9 @@ export class QueryBuilderState {
     this.queryTextEditorState = new QueryTextEditorState(this);
     this.queryUnsupportedState = new QueryBuilderUnsupportedState(this);
     this.mode = queryBuilderMode;
+    this.observableContext = new ObserverContext(
```
Comment (akphi) [inline packages/legend-query/src/stores/QueryBuilderState.ts:266]: :tada: exactly how I imagine it.

File: packages/legend-query/src/stores/QueryBuilderState.ts [hunk 3]
```diff
@@ -382,6 +391,10 @@ export class QueryBuilderState {
       const compiledLambda = guaranteeNonNullable(
         compiledValueSpecification.values[0],
       );
+      observe_ValueSpecification(
```
Comment (akphi) [inline packages/legend-query/src/stores/QueryBuilderState.ts:394]: Why don't we just observe `valueSpec` as we created it instead?

Comment (MauricioUyaguari) [reply packages/legend-query/src/stores/QueryBuilderState.ts:394]: Can you elaborate. We create it during the buiilder. I added it directly after building the value spec.

**Review by akphi — APPROVED** [review]

**Review by akphi — APPROVED** [review]

**Pull request merged**